### DISCUSSION TITLE
RP2040 :-assertion "ep->active" failed: file "/Users/babumongopi/r-pi/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040/hcd_rp2040.c", line 164, function: hw_trans_complete

### DISCUSSION THREAD

**Discussion by BabumonGopi** [hathach/tinyusb#1418] [Q&A]
Hi,
      I am using RP2040 as MSC host. Working with VS code on Mac. But after building with debug, the following error message is coming.
assertion "ep->active" failed: file "/Users/babumongopi/r-pi/pico-sdk/lib/tinyusb/src/portable/raspberrypi/rp2040/hcd_rp2040.c", line 164, function: hw_trans_complete
This is only for MSC app. HID and CDC are woking fine. When compiling with release mode, the controller is hanging and no response when connecting a MSC device (Flash drive). Any solution for this ? Anybody got success with RP2040 controller with MSC app ?

**Comment by hathach** [comment]
msc host on rp2040 is not supported yet, there are a couple of pending PRs to add support for it. But i haven't got time to test them out.

**Reply by BabumonGopi** [reply]
Thank you for your reply. Hope you will inform when it is ready to test.
